utils/blog/fetch-api.tsx

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import qs from "qs";
+import { getStrapiURL } from "./api-helpers";
+
+export async function fetchAPI(
+	path: string,
+	urlParamsObject = {},
+	options = {}
+) {
+	try {
+		// Merge default and user options
+		const mergedOptions = {
+			next: { revalidate: 60 },
+			headers: {
+				"Content-Type": "application/json",
+				Authorization: `Bearer ${process.env.NEXT_PUBLIC_STRAPI_API_TOKEN}`,
+			},
+			...options,
+		};
+
+		// Build request URL
+		const queryString = qs.stringify(urlParamsObject);
+		const urlPath = queryString
+			? `/api${path}?${queryString}`
+			: `/api${path}`;
```

Comment:
consider using our own `url_utils.ts` alternative for this.

Reply:
Going to continue using qs for now, using url_utils gave me an error and I didn't want to invest more time as this is running smoothly